Core/automation/lib/python/core/metadata.py

```diff
@@ -0,0 +1,193 @@
+
+def _resolveType(value):
+    '''Attempts to resolve the type of the metadata value.
+    It will return the value as the python type if possible,
+    otherwise will return value as string'''
+    if str(value).lower().strip() == "true":
+        return True
+    elif str(value).lower().strip() == "false":
+        return False
+    else:
+        # attempt to cast to int
+        try: return int(str(value))
+        except ValueError: pass
+        # attempt to cast to float
+        try: return float(str(value))
+        except ValueError: pass
+        # not a number
+        return str(value)
+
+class item_metadata(MutableMapping):
+    def __init__(self, item_name):
+        '''This is a manager class that can be used to manage multiple
+        metadata namespaces for a single item from one place.
+        - This class behaves like a dict of "metadata_namespace" objects.
+        - Namespaces can be added by name with "add_namespace(name)"
+        (attempting to add a namespace that already exists will raise a KeyError)'''
+        self._item_name = item_name
+        self._namespaces = {}
+
+    def __iter__(self): return iter(self._namespaces)
+    def __len__(self): return len(self._namespaces)
+    def __setitem__(self, key, value): raise AttributeError("Direct assignment to keys not possible")
+    def __getitem__(self, key): return self._namespaces[key]
+    def __delitem__(self, key): self.delete_namespace(key)
+
+    def add_namespace(self, name):
+        '''Creates a new namespace instance and loads it from the registry
+        if it exists.'''
+        if name in self._namespaces: raise KeyError
+        else: self._namespaces[name] = metadata_namespace(self._item_name ,name)
+
+    def delete_namespace(self, name, remove=False):
+        '''Deletes the specified namespace from this object. It will not
+        remove the namespace from openHAB unless the "remove" flag is set'''
+        if name not in self._namespaces: raise KeyError
+        else:
+            if remove: self._namespaces[name].remove
+            self._namespaces.pop(name, None)
+
+
+class metadata_namespace(MutableMapping):
+    def __init__(self, item_name, name, value=None, configuration={}, load=True, save=False):
+        '''Item metadata namespace
+        If "load" is set the namespace will be loaded from the registry. If it 
+        exists "value" and "configuration" will be overwritten if provided. 
+        If "save" is set the namespace will be written immediately to the 
+        registry. This will overwrite the namespace if it exists already. 
+        If the "load" flag is set the "save" flag will be ignored.'''
+        log.debug("Metadata: Initializing namespace object for item '{item}' namespace '{namespace}'".format( \
+            item=item_name, namespace=name))
+        self._item_name = item_name
+        self._name = name
+        self._keyUID = MetadataKey(self._name, self._item_name)
```

Comment:
_key_uid

Reply:
Oops right beside some lower_case ones too lol. I think I did this while testing and reflexively used camel case since it's my go to